doc/SONiC_static_route_hdl.md

```diff
@@ -0,0 +1,422 @@
+
+## 3.7 FRR Configuration Support
+### 3.7.1 Configuration Mapping to FRR
+Bgpcfgd daemon will be used to forward configurations stored in STATIC_ROUTE table to FRR staticd daemon. It will subscribe to listen to STATIC_ROUTE table and if there is data update, it will convert associated data to FRR vtysh command request and send to FRR daemon to configure static route on Linux kernel.
+#### 3.7.1.1 Table entry to command mapping
+FRR vtysh command is composed with VRF/IP_prefix and nexthop data fields in STATIC_ROUTE table entry
+#### FRR command syntax
+```
+configure terminal
+vrf <src_vrf>
+[no ]ip route <ip_prefix> [<nexthop_ip>] [<interface>] [<distance>] [nexthop-vrf <nh_vrf>]
+```
+#### Mapping from table entry to FRR command arguments
+|Parameter Name|Table Entry Key or Field|Type|Default or Null Value|
+|:---:|:-----------:|:------------------:|-----------------------------------|
+|src_vrf|first entry key*|optional for default VRF|"default"|
+|ip_prefix|second entry key*|mandatory|-|
+|nexthop_ip|entry field **nexthop****|optional if ifname is set or blackhole is true |"0.0.0.0" or "::"***|
+|interface|entry field **ifname****|optional if nexthop_ip is set or blackhole is true |""|
+|distance|entry field **distance****|optional|"0"|
+|nexthop_vrf|entry field **nexthop-vrf****|required for route-leak case else optional |""|
+|blackhole|entry field **blackhole***|required for blackhole next-hop else optional | false|
+
+Note:
+- *If table entry key contains only one item, it should be prefix_ip and src_vrf will be used with default value "default".
+- **Each argument uses one item of corresponding list in table entry. If the item in list is "Null" value, this optional argument will not be added to mapped FRR commnand.
+- ***The "Null" value of nexthop_ip should be chosen based on the address family of ip_prefix
+
+#### Example of table entry and correspoinding FRR runnning config
+Data in STATIC_ROUTE table:
+```
+127.0.0.1:6379[4]> hgetall STATIC_ROUTE|Vrf-test|1.1.1.0/16
+1) "nexthop@"
+2) "2.2.2.2,0.0.0.0"
+3) "distance@"
+4) "10,20"
+5) "ifname@"
+6) "Ethernet0,Ethernet4"
+7) "nexthop-vrf@"
+8) "Vrf-BLUE,Vrf-RED"
+9) "blackhole@"
+10) "false,false"
+```
+FRR running config:
+```
+!
+vrf Vrf-test
+ ip route 1.1.0.0/16 2.2.2.2 Ethernet0 10 nexthop-vrf Vrf-BLUE
+ ip route 1.1.0.0/16 Ethernet4 20 nexthop-vrf Vrf-RED
+ exit-vrf
+!
+```
+### 3.7.2 Configuration Reload
+All static route configurations were persistently stored in config DB STATIC_ROUTE table. After BGP container restarts, the configuration in DB needs to be re-applied. This configuration reload is done by generating staticd.conf file before FRR staticd daemon starts. A jinja template file will be used to map table entries to fill in staticd.conf file. The generated conf file is loaded by FRR daemon to configure static routes to system.
+
+# 4 Flow Diagrams

```

Comment:
We would need one or two flow diagrams to capture the end-to-end flow. You can refer VRF or Vxlan HLDs for the flow example

Reply:
okay.